### ISSUE TITLE
Async migrations

### ISSUE THREAD

**Issue by iNikem** [flyway/flyway#950]
Some migrations can effectively be run totally asynchronously relatively to other migrations. E.g. creating of index. It would be very nice to have a possibility to mark some migrations as async and not to block the whole process on them.

**Comment by axelfontaine** [comment]
I agree this can be valuable. It is not an immediate priority, but let's keep this open and see if it gathers traction.

Cheers
Axel

**Comment by anirudha13** [comment]
+1 for asynchronous migrations. We are using flyway with phoenix and some of our index creations take quite a long time.

**Comment by 6palace** [comment]
+1 we've also been affected by the lag on indexing extremely large tables, would love an async option to avoid downtime.

**Comment by f1xmAn** [comment]
Agree, that's valuable feature for us too.

**Comment by codersbrew** [comment]
Please implement :)

**Comment by aayushbhasin** [comment]
Would be an extremely useful feature.

**Comment by evanye** [comment]
+1

**Comment by ileshko** [comment]
+1!!

**Comment by codeyq** [comment]
+1!

**Comment by kakuism** [comment]
+1

**Comment by gvdenbro** [comment]
+1

**Comment by octavian-nita** [comment]
+1

Imagine that we might use Flyway's mechanisms to trigger the update of other, non/No-SQL repos such as ElasticSearch or even to send messages to various messaging/streaming infrastructures such as Kafka.

While Java migrations help a great deal since we can implement pretty much whatever we want (including spawning async jobs), native Flyway support would make things a breeze.

And... elaborating on this topic... I'm pretty much starting to see Flyway as a generic update mechanism and not only an SQL-related one. I am not sure whether this idea sprung before but a more decoupled, "update engine"-like mechanism would be great, imho.

**Comment by MikielAgutu** [comment]
@octavian-nita 
> And... elaborating on this topic... I'm pretty much starting to see Flyway as a generic update mechanism and not only an SQL-related one. I am not sure whether this idea sprung before but a more decoupled, "update engine"-like mechanism would be great, imho.

In Flyway 7 we shipped a new feature that will help with that: https://flywaydb.org/documentation/migrations#script-migrations

**Comment by detec** [comment]
+1

**Comment by sriram-au** [comment]
+1

**Comment by mackec1** [comment]
+1

**Comment by christrotteradhoc** [comment]
+1

**Comment by silvertonsg** [comment]
+1

**Comment by Jojoooo1** [comment]
+ 1

**Comment by rajeshhereforyou** [comment]
+1

**Comment by spark59** [comment]
+1

**Comment by EarthCitizen** [comment]
We ran into an issue in which our containers were failing health checks due to index creations taking too long. Async migration tasks are absolutely essential for embedded Spring Boot start-up migration.

**Comment by aprabhat** [comment]
@EarthCitizen I ran into same issue of POD getting restarted. Have you found any workaround for this?

**Comment by EarthCitizen** [comment]
@aprabhat In our case, they were index creations, and we created our Flyway scripts with "if not exists" for the indexes. Therefore, once they were created on the DB, the DB continued to work on them after the connection was lost when the container was restarted, and the scripts were marked as complete on the reboot due to the indexes already existing on the DB.

Basically, we got lucky.

**Comment by EarthCitizen** [comment]
> I agree this can be valuable. It is not an immediate priority, but let's keep this open and see if it gathers traction.

@axelfontaine This issue appears to have traction. There is a clear community need expressed here.

**Comment by aprabhat** [comment]
> @aprabhat In our case, they were index creations, and we created our Flyway scripts with "if not exists" for the indexes. Therefore, once they were created on the DB, the DB continued to work on them after the connection was lost when the container was restarted, and the scripts were marked as complete on the reboot due to the indexes already existing on the DB.
> 
> Basically, we got lucky.

I appreciate your response. In my case, I require the addition of a column. To achieve this, I implemented a workaround by creating a stored procedure that checks whether the column already exists. If the POD is terminated before successfully recording the entry in the schema version table, the subsequent POD will execute the same script. If the new column has already been created, the stored procedure will skip the "add column" step, and the new POD will successfully record the entry in the schema version table.

**Comment by tomekrojek** [comment]
+1

**Comment by Siddharth-Srinivas** [comment]
+1
